File: app/src/main/java/com/hedgehog/gdzietabiedra/anim/AnimationUtils.java
package com.hedgehog.gdzietabiedra.anim;

import android.animation.Animator;
import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.animation.PropertyValuesHolder;
import android.annotation.TargetApi;
import android.os.Build;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.animation.AccelerateInterpolator;
import android.view.animation.DecelerateInterpolator;
import android.view.animation.OvershootInterpolator;

import com.daimajia.androidanimations.library.Techniques;
import com.daimajia.androidanimations.library.YoYo;

/**
 * Created by Windows on 04-03-2015.
 */
public class AnimationUtils {

    @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
    public static void animateFadeIn(RecyclerView.ViewHolder holder, boolean goesDown) {

        if(Build.VERSION.SDK_INT  > Build.VERSION_CODES.ICE_CREAM_SANDWICH){
            int holderHeight = holder.itemView.getHeight();
            int holderWidth = holder.itemView.getWidth();
            View holderItemView = holder.itemView;
            holderItemView.setPivotY(goesDown ? 0 : holderHeight);
            holderItemView.setPivotX(holderWidth / 2);
            AnimatorSet animatorSet = new AnimatorSet();
            ObjectAnimator animatorScaleX = ObjectAnimator.ofFloat(holderItemView, "scaleX", .2f, 1f);
            ObjectAnimator animatorScaleY = ObjectAnimator.ofFloat(holderItemView, "scaleY", .2f, 1f);
            ObjectAnimator animatorAlpha = ObjectAnimator.ofFloat(holderItemView, "alpha", 0f, 1f);
            animatorAlpha.setInterpolator(new AccelerateInterpolator(1.5f));
            animatorSet.playTogether(animatorAlpha, animatorScaleX, animatorScaleY);
            animatorSet.setDuration(500).setInterpolator(new DecelerateInterpolator(1.1f));
            animatorSet.start();
        }
    }
}

File: app/src/main/java/com/hedgehog/gdzietabiedra/anim/AnimationUtils.kt
package com.hedgehog.gdzietabiedra.anim

import android.animation.AnimatorSet
import android.animation.ObjectAnimator
import android.annotation.TargetApi
import android.os.Build
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.animation.AccelerateInterpolator
import android.view.animation.DecelerateInterpolator

/**
 * Created by Windows on 04-03-2015.
 */
object AnimationUtils {

    @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
    fun animateFadeIn(holder: RecyclerView.ViewHolder,
                      goesDown: Boolean) {

        val holderHeight = holder.itemView.height
        val holderWidth = holder.itemView.width
        val holderItemView = holder.itemView
        holderItemView.pivotY = (if (goesDown) 0 else holderHeight).toFloat()
        holderItemView.pivotX = (holderWidth / 2).toFloat()
        val animatorSet = AnimatorSet()
        val animatorScaleX = ObjectAnimator
                .ofFloat(holderItemView, "scaleX", .2f, 1f)
        val animatorScaleY = ObjectAnimator
                .ofFloat(holderItemView, "scaleY", .2f, 1f)
        val animatorAlpha = ObjectAnimator
                .ofFloat(holderItemView, "alpha", 0f, 1f)
        animatorAlpha.interpolator = AccelerateInterpolator(1.5f)
        animatorSet.playTogether(animatorAlpha, animatorScaleX, animatorScaleY)
        animatorSet.setDuration(500).interpolator = DecelerateInterpolator(1.1f)
        animatorSet.start()
    }
}
